### ISSUE TITLE
[HOLD for payment 2022-02-16] Group creation - Search box is not cleared after selecting a user

### ISSUE THREAD

**Issue by isagoico** [Expensify/App#6872]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Shift+cmd+k or :green-plus-button: to start new group chat
2. Start typing name of user
3. Press enter to select user or click on user's name with cursor
4. Observe text from search for user is still in search box.

## Expected Result:
Search box should clear after selecting a user. 

## Actual Result:
Search box is not cleared after selecting a user

## Workaround:
User can delete the content in the search box. 

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 1.1.23-0

**Reproducible in staging?:** Yes 
**Reproducible in production?:** Yes 

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/44479856/147013069-34733ba5-05c7-4734-9912-1955c15b2290.mp4

**Expensify/Expensify Issue URL:**

**Issue reported by:** @mallenexpensify 
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1640130861454000

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @MariaHCD (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @kadiealexander (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by sobitneupane** [comment]
**Proposal**

Here, on selecting a row the searchValue is changed(change of state) to ' ' and on deselecting a row(i.e, removing selected member) searchValue remains as it is, i.e,
`searchValue: isOptionInList ? prevState.searchValue : ' '`

But, the changed searcheValue is not reflected from BaseExpensiTextInput as instead of using this.props.value while calling TextInputWithName Component,  this.value is being used.

So, changing 
` <TextInputWithName
         ----
         value={this.value}
         ---          
/>`
to
` <TextInputWithName
         ----
         value={this.props.value}
         ---          
/>`
in BaseExpensiTextInput will solve the issue.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @jliexpensify (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by kadiealexander** [comment]
Reassigning `external` as I'll be ooo for the next 2 weeks, thanks @jliexpensify!!

**Comment by jliexpensify** [comment]
Posted!

Internal - https://www.upwork.com/ab/applicants/1474202959761481728/job-details
External - https://www.upwork.com/jobs/~01eb6b3ea2ed5a0a26

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @parasharrajat (`Exported`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @pecanoro (`Exported`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by hiteshagja** [comment]
@jliexpensify 

Here is my proposal for this issue

### **Proposal**
I have added this new else if block
`else if (!this.props.value) {
      this.input.focus();
      this.setState({isFocused: true});
    } `

in `componentDidUpdate` method.

Updated `componentDidUpdate`  code
![image](https://user-images.githubusercontent.com/5493716/147466205-20326aed-fd56-4b10-86a9-4903867b402e.png)


### **Videos**
**Android**

https://user-images.githubusercontent.com/5493716/147464959-9199e507-b07c-44c6-965f-5eedc73130e1.mp4


**Web**

https://user-images.githubusercontent.com/5493716/147465385-aef1593b-ea69-473f-9b58-39ed7d4dd57a.mov



Does this make sense?

**Comment by parasharrajat** [comment]
@sobitneupane Are you suggesting to make the ExpensiTextInput as controlled?

**Comment by sobitneupane** [comment]
@parasharrajat Now, BaseExpensiTextInput is structured as a controlled component but update of on change text in being maintained in class variable. So, though the state of textinput was updated from NewChatPage.js considering BaseExpensiTextInput to be a controlled component, the effect is not reflected. So by using state instead of class variable, we can make BaseExpensiTextInput controlled.

**Comment by parasharrajat** [comment]
Thanks, @sobitneupane for clarification. I think It is fine to do. But I would like to discuss this in slack as this would affect all inputs in the app.

If we do this, there are a few other places where we have to do a small clean-up. I will point out those once we moved with your solution.

**Comment by sobitneupane** [comment]
Okay @parasharrajat . Should I message you in slack?

**Comment by parasharrajat** [comment]
Sorry, I meant I will discuss this on the public channel.

**Comment by parasharrajat** [comment]
https://expensify.slack.com/archives/C01GTK53T8Q/p1640793789214900

**Comment by parasharrajat** [comment]
Thanks for waiting on this. Unfortunately, keeping in mind that controlled input has performance issues, I can't accept the proposal that makes it controlled. 

@sobitneupane Do you have any other solution to fix it by making it controlled? It is fine even if it involves refactoring the BaseTextInput.

**Comment by sobitneupane** [comment]
To clarify further, as we are storing the value in class variable instead of a state, value of input is being changed or controlled but not updated in the react component. The changed value is passed from OptionsSelector to BaseExpensiTextInput. So, If we select two options for the same search value, for the second selection, the input value is changed to ' ' as the input value was updated to ' ' in previous step.

The other way around is to use this.props.value in place of this.value in BaseExpensiTextInput.

```
<TextInputWithName  
         ----  
         value={this.props.value}  
         ---            
/>
```

**Comment by parasharrajat** [comment]
Well, I understood your proposal. I will get back to you.

**Comment by parasharrajat** [comment]
Ok. Sorry for the delay. We are not ready to make BaseExpensiInput Controlled. @sobitneupane I understood your proposal but we can't do that do you have any other approach to solve this issue. 
Can we try with `ref.clear()` as the last resort or trigger an update on the input so that it clears the input automatically?
Or Do you think of a way so that `BaseExpensiInput` can be improved by making it controlled? 

I would prefer `Do you think of a way so that `BaseExpensiInput` can be improved by making it controlled? `

**Comment by sobitneupane** [comment]
We can clear the input when the row is selected by using `ref.clear()` in OptionsSelector.js. 

```diff
selectRow(option) {
        if (this.props.shouldFocusOnSelectRow) {
            this.textInput.focus();
        }
+        this.textInput.clear();
        this.props.onSelectRow(option);
    }
```

But for this to work we should also remove `value={this.value}` from BaseExpensiTextInput.js. And I don't think it will have major impact on any component using ExpensiTextInput as it is expected to be uncontrolled by other components as well.
```diff
<TextInputWithName
          ref={(ref) => {
              if (typeof this.props.innerRef === 'function') { this.props.innerRef(ref); }
              this.input = ref;
          }}
          // eslint-disable-next-line
          {...inputProps}
-         value={this.value}
          placeholder={(this.state.isFocused || !this.props.label) ? this.props.placeholder : null}
          .
          .
          .
 />
```

**Comment by parasharrajat** [comment]
> But for this to work we should also remove value={this.value} from BaseExpensiTextInput.js. And I don't think it will have major impact on any component using ExpensiTextInput as it is expected to be uncontrolled by other components as well.

Could you please explain it to me Why?

**Comment by sobitneupane** [comment]
> But for this to work we should also remove value={this.value} from BaseExpensiTextInput.js.

It should be removed because value = {this.value} will reset the value if not removed. 
> And I don't think it will have major impact on any component using ExpensiTextInput as it is expected to be uncontrolled by other components as well.

And It should not impact other components using it because this value is being used to take input from the parent component but as we are using class variable instead of state variable, value is not being updated.

**Comment by parasharrajat** [comment]
So if we remove this how are we supposed to pass the value to `ExpensiTextInput`. just trying to understand your idea.

**Comment by sobitneupane** [comment]
No, we won't be able to pass value from parent component to `ExpensiTextInput` (as we are expecting it to be uncontrolled).

**Comment by parasharrajat** [comment]
Sorry for the delay, I will share this internally to get more eyes.

**Comment by K4tsuki** [comment]
#### proposal

The problem is in this line

https://github.com/Expensify/App/blob/13b7b0b1945989379ab63f098ee2f480e1b1ba4b/src/components/TextInput/BaseTextInput.js#L183

value should be `this.props.value` and not `this.value`. Most recent value for textinput is in `this.props.value` and not in `this.value`. There is additional steps to set `this.value`.

**Comment by K4tsuki** [comment]
#### proposal

The problem is in this line

https://github.com/Expensify/App/blob/13b7b0b1945989379ab63f098ee2f480e1b1ba4b/src/components/TextInput/BaseTextInput.js#L183

value should be `this.props.value` and not `this.value`. Most recent value for textinput is in `this.props.value` and not in `this.value`. There is additional step to set `this.value`.

set value for `this.value` in `compponentDidUpdate`:

https://github.com/Expensify/App/blob/13b7b0b1945989379ab63f098ee2f480e1b1ba4b/src/components/TextInput/BaseTextInput.js#L55

This will not trigger render to render the text input;

Or there is other method: by calling `this.setValue` inside `componentDidUpdate`. But  something weird happen in the group `OptionSelector`

**Comment by K4tsuki** [comment]
How about changing below line with `this.setValue(this.props.value)` in:

https://github.com/Expensify/App/blob/991504fd8aefe8674f045bca11ae3c31f87337b2/src/components/TextInput/BaseTextInput.js#L55

And all `this.value.length` check condition must be coupled with `this.value`. So it will be `this.value && this.value.length`.

In this:

https://github.com/Expensify/App/blob/991504fd8aefe8674f045bca11ae3c31f87337b2/src/components/TextInput/BaseTextInput.js#L103

and this:

https://github.com/Expensify/App/blob/991504fd8aefe8674f045bca11ae3c31f87337b2/src/components/TextInput/BaseTextInput.js#L115

<br />
Also, doubled row option that you see after selecting a user (after this change) might be related to other bug about doubled option in group search.

The bug appear if you enter a text to search and then delete character with backspace. Happen in production.

**Comment by brianmuks** [comment]
**Proposal**
I have noticed that the issue is caused when the value is changed programmatically from outside. The input field needs to be focused for the update to take effect. Notice from the video that when the  `searchValue` is changed
https://github.com/Expensify/App/blob/0bce09177d2aed8c5e377bc758f5241dd22886eb/src/pages/NewChatPage.js#L188
The input field does not get updated. Yet when you navigate   away from the app  and back the input field automatically gets updated with previous changes.  So placing 
```
        this.input.focus();
```
below  
https://github.com/Expensify/App/blob/0bce09177d2aed8c5e377bc758f5241dd22886eb/src/components/TextInput/BaseTextInput.js#L55
does the trick . 

**Video**
 https://www.awesomescreenshot.com/video/6987681?key=c7d082e190c84cc093e339551e808522

**Comment by pecanoro** [comment]
@brianmuks That's interesting!

@parasharrajat Was there a conclusion in the discussion?

**Comment by parasharrajat** [comment]
@pecanoro No. Nobody seems to be interested 😅. Could you please reshare the thread in the Channel?

**Comment by parasharrajat** [comment]
Okay. There is no update on slack after pushing the discussion for several days. It seems to me that it is time to take the matter in my :open_hands: . 

@K4tsuki you have an interesting solution https://github.com/Expensify/App/issues/6872#issuecomment-1013060140. But let's demisify this. 
1. we send a value via props.
2. value is received as ComponentDidUpdate.
3. SetValue change `this.value` and you again push an update to the parent component via OnChange event. 
4. It means we receive a value from the parent and then you again send an update to the parent. 
5. Let's say a parent is passing the value from state and when the user types, it would update its state. Right? So onChange it updates the state. Now, it will send the value to TextInput via value prop and then the parent will do point 3 and loop continues. 

Am I missing something? 

---

At this stage, I would prefer to set the `value` to `this.props.value` and clean up the component. Thus I like @sobitneupane 's proposal. 

@pecanoro What do you think?

**Comment by K4tsuki** [comment]
Please wait...

**Comment by K4tsuki** [comment]
Let's take a look at real usage here:

https://github.com/Expensify/App/blob/07582967349266b430103b11a94839bf40d8bccb/src/pages/settings/Profile/ProfilePage.js#L217-L224

Component `FullNameInputRow` inside `ProfilePage` use this 'TextInput' component.

There is `FirstName` text input value set by `this.state.firstName`. 
Also there is `onChangeFirstName` that listen to `onChangeText `on first name that will `setState` of `firstName` each time the `onChangeText` being called. 
This is just like what you described.

<br />



The data flow when user type a character inside `firstName` textInput is:
1. `onChangeFirstname` being called
2. Profile page getting rendered, because state of `firstName` changed inside this function
3. `BaseTextInput` rendered/updated 
4. on `componentDidUpdate` of `BaseTextInput` passing check on early return and then `setValue` being called => `onChangeFirstname` called again, back to step 1 until 3 and then early return happen in:
https://github.com/Expensify/App/blob/07582967349266b430103b11a94839bf40d8bccb/src/components/TextInput/BaseTextInput.js#L51-L53

**Comment by parasharrajat** [comment]
yeah, I got your point but I leaning more toward `I would prefer to set the value to this.props.value`.

**Comment by K4tsuki** [comment]
It will become controlled.
For controlled `TextInput` you can use `import {TextInput} from 'react-native';` Not this `TextInput` inside element. 
Please respect decision from @marcaaron and other in slack.

**Comment by parasharrajat** [comment]
There is no conclusion on slack yet.

AFAIK, our TextInput is already controlled. Using `value` prop makes TextInput controlled. It does not matter if it is an instance or state property.  https://expensify.slack.com/archives/C01GTK53T8Q/p1642096772440500?thread_ts=1640793789.214900&cid=C01GTK53T8Q

> Please respect decision from marcaaron and others in slack.

It is my opinion and I have asked the concerned person. I would be thankful if you be patient.

**Comment by K4tsuki** [comment]
> I believe there are performance issues related to having inputs be controlled and that’s why we are planning to standardize on uncontrolled.

This is what @marcaaron said in slack.

**Comment by pecanoro** [comment]
Ok, let me revive the conversation in Slack since I need the opinion of the most knowledgeable React Native Expensify engineers, I am stronger on the back-end side so I can't see all the ramifications of making the component controlled.

**Comment by K4tsuki** [comment]
> Or we can explicitly clear the input 

That is what Marcaaron emphasized in slack.
So we need a reference to the `TextInput` call `clear()` method. Or using `setNativeProps ({text: ''})`

**Comment by parasharrajat** [comment]
Yeah, that can be an option. Maybe we can remove the value prop and use `defaultValue` instead of it with this if that works.

**Comment by marcaaron** [comment]
Ok, seems like this needs a push so I took a deeper look.

Yes, the input is controlled (sort of..) we are doing a lot of weird stuff and the custom `TextInput` component is pretty complex so I think we're both kind of right, sorry for the confusion @parasharrajat.

Given where we are at, I think it is fine for the `TextInput` to work in a controlled way wherever it is used. But it's not working correctly. We are missing a step which is that if `props.value === ''` then we must programmatically clear the input.

I'm not sure if input _focusing_ is on the right. This issue is only mentioning the input _clearing_ not suggesting we bring the focus back to the input. That seems like a separate issue maybe or working because of a side-effect.

So, the issue is that the `TextInput` does not clear if the `value` changes and we have passed an empty string for the value.
Why does it happen? The answer is...

We only handle the change in value when the component updates. Which happens after the render. So it is a controlled component, but not really as we might expect as the value is only passed after the component renders. Here's the typical case of adding some text to the input...

- User changes text -> 
- `TextInput` value changes -> 
- change handler calls `this.props.onChange` ->
- parent passes a new `props.value` ->
- that value DOES NOT get passed to the component in the `render()` but the value has changed on screen ->
- `componentDidUpdate()` handles the new value and sets `this.value` ->
- that does not trigger the component to re-render (but again, the value has changed on screen so all is fine and the value we see is synced with the value in `this.value`)

Let's go through the same flow, but test selecting a user

- User adds a participant ->
- `props.value == ''` is passed to `TextInput` ->
- `TextInput` `render()`
- `TextInput` sees that the value changed and update `this.value` to an empty string ->
- But the value on-screen does not change... 💥 

It doesn't change because the component is not re-rendering (required to pass `this.value` to the underlying `TextInput`). To make this work correctly I think we can just do something like

```diff
diff --git a/src/components/TextInput/BaseTextInput.js b/src/components/TextInput/BaseTextInput.js
index c95856b0b..239766c5f 100644
--- a/src/components/TextInput/BaseTextInput.js
+++ b/src/components/TextInput/BaseTextInput.js
@@ -53,6 +53,9 @@ class BaseTextInput extends Component {
         }

         this.value = this.props.value;
+        if (this.props.value === '') {
+            this.input.clear();
+        }

         if (this.props.value) {
             this.activateLabel();
```

If that works OK, can we do it and follow up to improve this component later?

**Comment by marcaaron** [comment]
@K4tsuki seems like you were the closest to this solution. Everyone else agree with that and are we good to assign them?

**Comment by parasharrajat** [comment]
Thanks for investigating @marcaaron. I am glad that we have a mutual agreement on what should be done.  

> follow up to improve this component later?

I am working on cleaning up our TextInputs https://github.com/Expensify/App/issues/6584. I can add another TODO there.

yeah, @K4tsuki was closet. He suggested using `So we need a reference to the TextInput call clear() method. Or using setNativeProps ({text: ''})` . 

I like the latest suggestion from Marc. 

cc: @pecanoro 

:ribbon: :eyes: :ribbon:  C+ reviewed

**Comment by K4tsuki** [comment]
Thank you for the great explanation @marcaaron.

@marcaaron this is works for `this.props.value == ''`, but for value other than = ''. It still doesn't work. 
I have tested in `NewChatPage` constructor, windows `setTiimeout` to change `searchValue` state and it is not changing.

In `componentDidMount` of BaseTextInput filter check should be:
`if (this.value === this.props.value)` in:

https://github.com/Expensify/App/blob/991504fd8aefe8674f045bca11ae3c31f87337b2/src/components/TextInput/BaseTextInput.js#L51

This will filter out changeText by user typing, because `onChangeText` of BaseTextInput is setting `this.value` and calling consumer `onChangeText`. So on `componentDidUpdate` of `BaseTextInput`, `this.value` is same as `this.props.value` so we must filter it.

After ` this.value == this.props.value;` we could call  `this.input.setNativeProps({ text: this.props.value});`

<br />

If I am really assigned to this, I want to divide the reward with Brianmuks and Sobitneupane . I think proposals are close and we have been working hard for this (If I really assigned to this). 

------------------
@parasharrajat I am sorry I thought you are jumping to the conclusion to change `value={this.value}` into `value={this.props.value}`

**Comment by marcaaron** [comment]
>@marcaaron this is works for this.props.value == ''

Yes, when selecting a user the `props.value` is set to an empty string and we need the `TextInput` to clear

>but for value other than = ''. It still doesn't work.

What doesn't work exactly?

>I want to divide the reward with Brianmuks and Sobitneupane

That's very kind. Not sure anyone has asked this before, but can relay this feedback to the team.

**Comment by K4tsuki** [comment]
> What doesn't work exactly?

Other than empty string.

<br />

And shouldn't this check: 

https://github.com/Expensify/App/blob/991504fd8aefe8674f045bca11ae3c31f87337b2/src/components/TextInput/BaseTextInput.js#L51

be `if (this.value === this.props.value)`

because `this.value` changes when user is typing but doesn't change when consumer of `BaseTextInput` set the `props.value`.
and then we can call `this.input.setNativeProps({ text: this.props.value});`?

**Comment by parasharrajat** [comment]
Ah, I see your point now @K4tsuki. To clarify
1. When the user typed `acids`. 
2. On the Consumer(parent component) we blocked the update of `value` as we are only accepting numbers.  So value will be `''`.
3. Now on the TextInput the `this.value === acids` but `this.props.value === ''` and also the `prevProps.value` would be `''` thus componentDidupdate will early return. 

but the textinput will keep `acids` as `this.value` is till `acids`. 

So you are suggesting we use `this.input.setNativeProps({ text: this.props.value});` to update this. I agree this should be used to update uncontrolled. 

But I tried and change is not spontaneous. textInput only update when we blur the focus from it.

**Comment by parasharrajat** [comment]
So we will have to force update. Do you see any other solution that problem? 
```diff
--- a/src/components/TextInput/BaseTextInput.js
+++ b/src/components/TextInput/BaseTextInput.js
@@ -46,13 +46,15 @@ class BaseTextInput extends Component {
         this.input.focus();
     }
 
-    componentDidUpdate(prevProps) {
+    componentDidUpdate() {
         // activate or deactivate the label when value is changed programmatically from outside
-        if (prevProps.value === this.props.value) {
+        if (this.value === this.props.value) {
             return;
         }
 
         this.value = this.props.value;
+        this.input.setNativeProps({text: this.value});
+        this.forceUpdate();
 
         if (this.props.value) {
             this.activateLabel();
 ```

**Comment by K4tsuki** [comment]
I test using

https://github.com/Expensify/App/blob/2cdb7c30be70766b3c006a67153665d895ba83ac/src/pages/NewChatPage.js#L188

Instead of `empty string` try to change in to `lorem ipsum` to be : `searchValue: isOptionInList ? prevState.searchValue : 'lorem ipsum',`

And it works without `forceUpdate()`

Where is it doesn't work? 
In the same new group chat page?

**Comment by parasharrajat** [comment]
Ok, then it's fine. we can leave the `forceUpdate()`.

**Comment by K4tsuki** [comment]
Similar to what you write above, but without `foceUpdate`

```diff
--- a/src/components/TextInput/BaseTextInput.js
+++ b/src/components/TextInput/BaseTextInput.js
@@ -46,13 +46,15 @@ 
    class BaseTextInput extends Component {
         this.input.focus();
     }
 
-    componentDidUpdate(prevProps) {
+    componentDidUpdate() {
         // activate or deactivate the label when value is changed programmatically from outside
-        if (prevProps.value === this.props.value) {
+        if (this.value === this.props.value) {
             return;
         }
 
         this.value = this.props.value;
+        this.input.setNativeProps({text: this.value});
 
         if (this.props.value) {
             this.activateLabel();

```

`(this.value === this.props.value)` this check is filters user typing in text input, so  we don't call `setNativeProps` needlessly,

**Comment by marcaaron** [comment]
>because this.value changes when user is typing but doesn't change when consumer of BaseTextInput set the props.value.

Hi, can we wrap this up, please? I tested the solution I shared and it seemed to work. Is there some problem with it and does it not fix the issue? Remember, we are trying to fix _this_ issue not some other issue or problem we may have identified.

What is the point of using `setNativeProps()`?

**Comment by K4tsuki** [comment]
@marcaaron If we solve this current issue then using your suggestion is best. 

It is used for other problem. I have tested it using `lorem ipsum` [above](https://github.com/Expensify/App/issues/6872#issuecomment-1024813005), then the search box doesn't display `lorem ipsum` after select a user. 
It means if `props.value` is supplied to `BaseTextInput` there is still a problem of not being displayed.

**Comment by marcaaron** [comment]
Sounds good! I'm not sure what other problem we are trying to solve, but if there is one feel free to report it.

**Comment by K4tsuki** [comment]
All right let's wrap this up and using @marcaaron suggestion above. 
Thank you @marcaaron

**Comment by jliexpensify** [comment]
I'm assuming we're hiring @K4tsuki ?

**Comment by parasharrajat** [comment]
Yeah, I agree. I faced the problem that @K4tsuki mentioned in my [PR](https://github.com/Expensify/App/pull/7318#issuecomment-1024804117)  and the [these](https://github.com/Expensify/App/issues/6872#issuecomment-1024806137)  reproduction steps will match with the issue that I faced on that PR. Thus I pointed out here as @K4tsuki found the cause of the issue. [This](https://github.com/Expensify/App/issues/6872#issuecomment-1024812272) solution fixes the issue on that PR and requires force update as well. 

As Marc said, we can just focus on fixing this GH issue and I will fix the other that I faced on m [PR](https://github.com/Expensify/App/pull/7318#issuecomment-1024804117) which will require similar solution  to https://github.com/Expensify/App/issues/6872#issuecomment-1024812272

**Comment by pecanoro** [comment]
Thank you for the discussions, everyone, it took me a bit to catch up but I got to learn a lot about this flow. Assigning @K4tsuki since we all agreed it seems to be the best approach and it seems like a good solution to the best of my knowledge.

**Comment by MelvinBot** [comment]
📣 @K4tsuki You have been assigned to this job by @pecanoro!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md) 📖

**Comment by jliexpensify** [comment]
@K4tsuki could you link me your Upworks profile please? Thanks!

**Comment by K4tsuki** [comment]
@jliexpensify I want  to divide the reward for the three of us: me, brianmuks and sobitneupane as I state above. So the reward  is divided into three. 
Here is my upwork profile:
https://www.upwork.com/freelancers/~01bdd9e3a6f15042bd

Thank you.

**Comment by marcaaron** [comment]
@K4tsuki Unfortunately, we don't really have a system to facilitate this. If you want to share the fee then that is up to you, but we can only hire one person to complete the job.

**Comment by K4tsuki** [comment]
@marcaaron all right, I will follow the system. 
And will contact sobitneupane and brianmuks later.
Thank you.

**Comment by jliexpensify** [comment]
@parasharrajat @pecanoro - just wondering: am I good to pay this one? It looks like there was a failure on the iOS end

![image](https://user-images.githubusercontent.com/30815269/154395083-0b6503db-4589-4443-b396-eb5e0a6923ca.png)

**Comment by parasharrajat** [comment]
Yeah, @jliexpensify It is deployed. That was some other glitch.

**Comment by jliexpensify** [comment]
Cheers, paying now:

- @K4tsuki $250
- @parasharrajat $250

Will close once done
